compiler/src/dmd/func.d

```diff
@@ -469,25 +469,27 @@ extern (C++) class FuncDeclaration : Declaration
     /****************************************************
      * Resolve forward reference of function signature -
      * parameter types, return type, and attributes.
+     * Params:
+     *  fd = function declaration
      * Returns:
      *  false if any errors exist in the signature.
      */
-    final bool functionSemantic()
+    static bool functionSemantic(FuncDeclaration fd)
```

Comment:
You've done the hard bit of making this static, why not just move the function to toplevel so that all callers remain unaffected (ufcs)?

Reply:
Because I am going to move it to a new module `functionsem.d`, but I want that move to be a cut&paste job for easy review. The idea is to separate the semantic code from the ast code so that dmd-as-library will be more tractable, we can get rid of the duplicate ast declarations, and we can move towards making dmd a language server.

Reply:
Yes, and when you move it to `functionsem.d` you'll have to change all the callers back to how they were originally.